source/Classes/1.DscResources/07.nxService.ps1

```diff
@@ -40,7 +40,7 @@ class nxService
         $currentState.State = $nxService.State
         $currentState.Controller = $this.Controller
 
-        if (-not $currentState)
+        if (-not $currentState.Name)
         {
             # Silently return if the service does not exist
             Write-Warning -Message ('Service ''{0}'' could not be found.' -f $this.Name)
```

Comment:
I know you didn't edit this part, but can we not return just a completely blank nxService in this case? (line 47)
Can we at least return a Reason that says the service does not exist or something?

Reply:
Before I make this change,

- This resource is written so that if the service doesn't exist, it's compliant.
- The test function is written to return true if there are 0 reasons.

I will have to change the implementation of test if I return a reason. Is that okay?

Reply:
So if I say that I want a service Enabled and in the Running state on the machine, but that service doesn't exist on the machine, then we are compliant?
I think as a customer that is not what I would want?
If the customer said that they want the service not Enabled or in like a Disabled state and it doesn't exist, then that seems compliant?
It does seem like this would be a breaking change, but as it is now I think it's broken to say that if the service does not exist, then the resource is always compliant?

Reply:
Thanks Katie. I updated this to return a reason in some cases. More context in the email I sent. Let me know if we need to discuss.